src/_pytest/debugging.py

```diff
@@ -289,6 +289,7 @@ def _enter_pdb(node, excinfo, rep):
     tw.sep(">", "entering PDB")
     tb = _postmortem_traceback(excinfo)
     rep._pdbshown = True
+    sys.last_type, sys.last_value, sys.last_traceback = excinfo._excinfo
     post_mortem(tb)
     return rep
```

Comment:
Initially I wanted this to use `Pdb.user_exception` instead, but that would require some adjustments of (mocking) tests.
`user_exception` would set `__exception__` into locals (but only type and value currently (https://bugs.python.org/issue36663)).